backend/src/main/kotlin/clarity/backend/entity/UserEntity.kt

```diff
@@ -26,14 +30,32 @@ class UserEntity() {
             val selectStatement = "SELECT * FROM User WHERE username = '${user.username}' AND password = '${user.password}'";
             val result = statement.executeQuery(selectStatement);
             if(result.next()) {
-                return UserWithId(
+                val lastLoggedIn = result.getString("last_logged_in")
+                val loginStreak = result.getInt("login_streak")
+                val currentDate = LocalDate.now()
+
+                // Check difference between last login date and today
+                val newLoginStreak = when (ChronoUnit.DAYS.between(currentDate, LocalDate.parse(lastLoggedIn, DateTimeFormatter.ISO_DATE)).toInt()) {
+                    0 -> loginStreak
+                    1 -> loginStreak + 1
+                    else -> 1
+                }
```

Comment:
Not sure w/ Kotlin. But will this cause an error with the default NULL value for `lastLoggedIn`? Or does getString() handle null properly?

Cause otherwise, we might get an error for LocalDate.parse(...) with lastLoggedIn being NULL.

Reply:
Oh right, yea that makes sense. Thx.